pkg/util/kubernetes/autoscalers/datadogexternal.go

```diff
@@ -184,17 +184,23 @@ func (p *Processor) updateRateLimitingMetrics() error {
 func NewDatadogClient() (*datadog.Client, error) {
 	apiKey := config.Datadog.GetString("api_key")
 	appKey := config.Datadog.GetString("app_key")
+	endpoint := config.Datadog.GetString("external_metrics_provider.endpoint")
 
 	if appKey == "" || apiKey == "" {
 		return nil, errors.New("missing the api/app key pair to query Datadog")
 	}
 
+	if endpoint == "" {
+		endpoint = config.GetMainInfraEndpoint()
```

Comment:
this will return for example `https://app.datadoghq.com` instead of `https://api.datadoghq.com`, is that OK?

Reply:
Good catch, that's not ok. I amended the changes, and also updated the PR description to match the new changes.